### PR TITLE
cxx-qt-lib: Add chrono and time support for {QDate,QDateTime,QTime}

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Requires #416

Related to #292

### CODE REVIEW

File: crates/cxx-qt-lib/src/core/qdate.rs [hunk 1]
```diff
@@ -86,3 +223,98 @@ unsafe impl ExternType for QDate {
     type Id = type_id!("QDate");
     type Kind = cxx::kind::Trivial;
 }
+
+#[cfg(feature = "chrono")]
+use chrono::Datelike;
+
+#[cfg(feature = "chrono")]
+impl From<chrono::NaiveDate> for QDate {
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qdate.rs:231]: I've gone with by value in this pull, elsewhere we thought it was more typical in Rust to by value. We should probably other API both internal and external to figure out if they should be by value or ref?

Eg here should it be as we end up copying internally anyway ?

**Review by LeonMatthesKDAB — COMMENTED** [review]
The chrono and time changes look good.
A minor improvement is possible on `use super::*;`, but that's it I think

File: crates/cxx-qt-lib/src/core/qtime.rs [hunk 2]
```diff
@@ -240,7 +266,7 @@ unsafe impl ExternType for QTime {
 
 #[cfg(test)]
 mod test {
-    #[cfg(feature = "chrono")]
+    #[cfg(any(feature = "chrono", feature = "time"))]
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply crates/cxx-qt-lib/src/core/qtime.rs]: Moved to the module

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**